dkg/nodesigs.go

```diff
@@ -50,32 +62,64 @@ func newNodeSigBcast(operatorAmt int, bcastComp *bcast.Component) nodeSigBcast {
 }
 
 // broadcastCallback is the default bcast.Callback for nodeSigBcast.
-func (n *nodeSigBcast) broadcastCallback(_ context.Context, _ peer.ID, _ string, msg proto.Message) error {
+func (n *nodeSigBcast) broadcastCallback(ctx context.Context, peer peer.ID, _ string, msg proto.Message) error {
 	response, ok := msg.(*dkgpb.MsgNodeSig)
 	if !ok {
 		return errors.New("invalid node sig type")
 	}
 
-	n.otherSigs <- response
+	sig := response.GetSignature()
+	msgPeerIdx := int(response.GetPeerIndex())
+
+	if (msgPeerIdx == n.nodeIdx.PeerIdx) || (msgPeerIdx < 0 || msgPeerIdx > n.operatorAmt) {
+		return errors.New("wrong peer index", z.Str("peer", peer.String()))
+	}
+
+	peerPubk, err := n.peers[msgPeerIdx].PublicKey()
+	if err != nil {
+		return errors.Wrap(err, "can't get peer public key", z.Str("peer", peer.String()))
```

Comment:
nit: keep wrapping error messages super concise. It isn't always required to say "this failed while doing X", since it is clear from the context that this is an error that stuff failed while doing X. So you can just do: `errors.Wrap(err, "get peer pubkey")`